scripts/recan_bundle_tdr.py

```diff
@@ -109,8 +113,11 @@ def content_length(content: JSON) -> int:
     return len(json.dumps(content).encode('UTF-8'))
 
 
-def random_uuid() -> str:
-    return str(uuid.uuid4())
+def deterministic_uuid(bundle_uuid: str, *names: str, sep='\0') -> str:
+    namespace = uuid.UUID(bundle_uuid)
+    for name in names:
+        reject(sep in name, 'Separator must not occur in name elements', sep, name)
```

Comment:
It's not a user error if this fires. It should be an assert and doesn't need a special message.

```suggestion
        assert sep not in name, (sep, name)
```